### PR TITLE
wire: remove code generator tool

### PR DESCRIPTION
**Pull request by zombiezen**
The marker function package remains where it is, but now aliases to the new Wire package. I will update the import paths in this repository in a separate PR so as to shake out any unforeseen issues that would be encountered by end-users as well.

I removed the content of the README and left behind a pointer to the new repository at https://github.com/google/wire. I fixed any dangling documentation links I could find.

Updates #828

/cc @vangent @clausti

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,8 +24,8 @@ and being able to run that code on any cloud you want, avoiding all the ceremony
 of cloud-specific authorization, tracing, SDKs and all the other code required
 to make an application portable across cloud platforms.
 
-Another key part of the project is a code generator called
-[Wire](https://github.com/google/go-cloud/blob/master/wire/README.md). It
+The project works with a code generator called
```
Comment (vangent) [inline README.md]: Maybe "works well with"?
I don't think that we want to imply that it requires Wire.

File: doc.go [hunk 2]
```diff
@@ -26,8 +26,8 @@ gcsblob.OpenBucket, s3blob.OpenBucket, or any other provider. Then,
 the blob.Bucket can be used throughout your application without worrying
 about the underlying implementation.
 
-This project also provides a code generator called Wire
-(https://github.com/google/go-cloud/blob/master/wire/README.md). It creates
+This project works with a code generator called Wire
```
Comment (vangent) [inline doc.go]: Same.

File: samples/wire/README.md [hunk 3]
```diff
@@ -405,12 +405,12 @@ There is support for [binding interfaces][interfaces], [binding
 values][values], as well as support for [cleanup functions][cleanup]. See the
```
Comment (vangent) [inline samples/wire/README.md]: Should this whole sample be moved to the `wire` repo? It seems to be all about wire.

Fine to do in a separate PR.

**Comment by vangent** [comment]
Tests failed. I suspect you need to update the proxy with the packages for `wire`.

Are we sharing the same proxy? If so, we'll need to update the instructions to always build wire as well as go-cloud. It'll probably be easier/better to have separate proxy buckets.

File: samples/wire/README.md (hunk 3 above)
Comment (shantuo) [reply samples/wire/README.md]: Agree to separate the sample, but we'll want some pointers in this repo to Wire as well.

File: .travis.yml [hunk 4]
```diff
@@ -23,7 +23,7 @@ install:
   # converting \n to \r\n.
   - "git config --global core.autocrlf input"
   - "git checkout -- ."
-  - "go install ./wire/cmd/wire"
+  - "go install github.com/google/wire/cmd/wire"
```
Comment (shantuo) [inline .travis.yml]: Are we generating code in the tests though?

Comment (zombiezen) [reply .travis.yml]: We're running `wire check` to ensure that there are no errors in the provider sets we give. See here: https://github.com/google/go-cloud/blob/87e273223b452e987ff654f016887ab80a6f9252/internal/testing/runchecks.sh#L38

File: README.md (hunk 1 above)
Comment (zombiezen) [reply README.md]: Done.

File: doc.go (hunk 2 above)
Comment (zombiezen) [reply doc.go]: Done.

File: samples/wire/README.md (hunk 3 above)
Comment (zombiezen) [reply samples/wire/README.md]: Agreed. Opened #836 to track removing the sample from this repository.

**Comment by zombiezen** [comment]
Yup, I forgot to update the proxy before. I have updated it now.

**Comment by zombiezen** [comment]
@shantuo, I'm blocked on your review.

**Comment by shantuo** [comment]
Discussed offline, agreed on removing from the current version.
